osfs/os.go

```diff
@@ -14,7 +14,7 @@ import (
 )
 
 const (
-	defaultDirectoryMode = 0o755
```

Comment:
Please add some tests to confirm the new behaviour - so that we don't have a regression in the future.